### DISCUSSION TITLE
Slash_commands and interactions

### DISCUSSION THREAD

**Discussion by jbaumgarten** [Rapptz/discord.py#8424] [Q&A]
Hi guys,

I have several servers where there are some slash commands to interact with it.
Some slash commands requires some parameters which are, I suppose, a kind of context.
Is it possible to use that commands with this library?

Some slash commands take channel member as argument

**Comment by DenverCoder1** [comment]
You can see some examples of Slash Commands in the examples folder:

https://github.com/Rapptz/discord.py/tree/master/examples/app_commands

Parameters to slash commands work the same as adding parameters to a Python function, you can use type annotations to make them take different types of input, for example `discord.Member` for members, `int` for numbers, etc.

```py
@client.tree.command()
@app_commands.describe(member='The member you want to get the joined date from')
async def joined(interaction: discord.Interaction, member: discord.Member):
    """Says when a member joined."""
    # The format_dt function formats the date time into a human readable representation in the official client
    await interaction.response.send_message(f'{member} joined {discord.utils.format_dt(member.joined_at)}')
```

You can see more info about the supported annotations here: https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.app_commands.Namespace

**Reply by jbaumgarten** [reply]
Thx for the answer.
Btw, this piece of code is to build a bot for your server, not to act as a client, right?

**Reply by DenverCoder1** [reply]
Yeah, the "client" referenced here is the interface that allows the bot user to connect to the Discord API. You can check the full examples and more details in the documentation.

**Reply by jbaumgarten** [reply]
ok but is it possible to dev a python client that act like a real user? I know you can send messages, ... but I want to automate some action which are slash commands using members

**Reply by DenverCoder1** [reply]
Impersonating a non-bot user is against the Discord TOS and not possible with this library.

As far as I know, automating the usage of slash commands is not currently possible not even as a bot user.

**Reply by jbaumgarten** [reply]
Yes I know it's against the Discord TOS but ... :)

**Reply by IAmTomahawkx** [reply]
You will not receive help from this repo on userbotting.

**Reply by jbaumgarten** [reply]
Ok...

**Reply by jbaumgarten** [reply]
Discum library give me what I want.
Thx anyway
